pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -334,6 +335,37 @@ impl InMemoryLayer {
                     newsize,
                     lsn
                 );
+
+                // If we are extending the relation by more than one page, initialize the "gap"
+                // with zeros
+                //
+                // XXX: What if the caller initializes the gap with subsequent call with same LSN?
```

Comment:
just for my own understanding, what would we see in args for these two calls to `put_page_version`?

Reply:
Something like this:

(relation size is 9 at the start, i.e. blk number 8 is the last block)
put_page_version(10, Lsn(123), ...); // extends the relation by two blocks, fills block 9 with zeros
put_page_version(9, Lsn(123), ...); // writes real data to block 9 at the same LSN

Reply:
Ahh I see. 
Maybe we could store if the page had been initialized empty for the current LSN